### PR TITLE
Move namespace utils from gProfiler @ 89fcf8e2717 and improve them

### PR DESCRIPTION
**Pull request by Jongy**
* Move namespace utils from gProfiler @ 89fcf8e2717
* Make `run_in_ns` re-raise the exception if one is raised in its thread
* FIx `ns_path` absolute check in `resolve_proc_root_links`

### CODE REVIEW

File: granulate_utils/linux/ns.py
Comment (YishaiZinkin) [inline granulate_utils/linux/ns.py:102]: Explain in a comment why you're doing that (i.e., that you want the exception to take place in the calling thread)

Comment (Jongy) [reply granulate_utils/linux/ns.py:102]: kk

File: granulate_utils/linux/ns.py
Comment (d3dave) [inline granulate_utils/linux/ns.py:82]: Extract to global along with the functions so that we don't `dlopen`, `dlclose`, `dlsym` all the time.

File: granulate_utils/linux/ns.py
Comment (d3dave) [inline granulate_utils/linux/ns.py:90]: `enum.IntFlag`

**Review by d3dave — CHANGES_REQUESTED** [review]

**Comment by YishaiZinkin** [comment]
Looks good AFAIC, just wait for @d3dave's approval

**Review by d3dave — APPROVED** [review]

**Pull request merged**
